### DISCUSSION TITLE
A serverless notebook experimentation with pyscript, tailwind, and markedjs

### DISCUSSION THREAD

**Discussion by nicolasgallandpro** [pyscript/pyscript#1832] [Show and tell]
Hello everyone, and thanks to the team for their excellent work on PyScript. 

As a dedicated Streamlit user, I wanted to experiment with creating a notebook using PyScript. The idea was to create a serverless notebook with interactivity and a developer-friendly experience. I used marked.js to convert the markdown into HTML and Tailwind CSS (along with its Typography and Forms plugins) for the rendering.

The experience was surprisingly straightforward: you write your text in Markdown, your Python code, you save it, and it works. It offers interactivity without the need for compilation (unlike Quarto) or a server (unlike Streamlit). The code is simple. I started using this to deliver one-time analysys to my clients, and it's nice.

You can see the result here : https://nicolasgallandpro.github.io/open_data_exploration/pyscript_demo/

I also wrote an article to explain my approach, the pros and the cons : https://medium.com/@nicogalland/a-simple-serverless-static-interactive-python-notebook-with-pyscrypt-marked-js-and-tailwind-css-ef40419345d1

![image](https://github.com/pyscript/pyscript/assets/61990668/67d7056d-c91e-4303-a2cb-914a02a0be77)

The code : 

![image](https://github.com/pyscript/pyscript/assets/61990668/4e6d11ee-ff83-4867-83df-1e9fd158a0f9)

The only real drawback I encountered is the inability to read remote Parquet files (soon ?)

**Comment by sskagemo** [comment]
Nice :)
Just a comment on the parquet-problem; I managed to solve it in my experiment with the Norwegian Business Register. I published a parquet-file on Google Cloud, and added it in the pyscript.toml-file, so it is downloaded as a requirement before any other code is executed. And then I can access it as a "local" file (i.e. the browsers local file system).

[See the code and solution here](https://pyscript.com/@aaa95cd8-38ea-429f-8399-087a472f9e7d/7b00bb5a-bc81-4ccb-b16b-b169aef6d39f/latest)

[I also wrote about it here in "Show and tell"](https://github.com/pyscript/pyscript/discussions/1609)

**Reply by nicolasgallandpro** [reply]
Thanks ! I'll try this

**Reply by sskagemo** [reply]
Good luck! Just forgot to mention that I had to solve the problem with cors-headers, that was why I had to publish the file through Google Cloud, and not GitHub, which was the initial idea.
